packages/grid/x-data-grid/src/hooks/utils/useGridApiMethod.ts

```diff
@@ -20,14 +20,10 @@ export function useGridApiMethod<
       return;
     }
     apiMethodsNames.forEach((methodName) => {
-      if (!privateApiRef.current.hasOwnProperty(methodName)) {
```

Comment:
The goal here was to have stable references to API methods.
For example, with this change, it's now impossible to `spy` on API methods, because the `spy` gets overridden with the new function reference (this is why the tests fail).

Reply:
Could we use `useEventCallback` on the methods instead ?

Reply:
Is there any reason other than testing for which we would like to keep the stable references?

Reply:
I've benchmarked this one, for the raw filtering time, this indirection degrades performance by 11% if we take the final results as the baseline.

Reply:
As far as I know, testing is the only reason to keep the stable references.

Reply:
Got it, I'll try to make it work without the trampoline then.